airflow/providers/microsoft/azure/hooks/azure_functions.py

```diff
@@ -0,0 +1,113 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Any
+
+import requests
+from azure.identity import ClientSecretCredential
+
+from airflow.providers.http.hooks.http import HttpHook
+from airflow.providers.microsoft.azure.utils import get_field
+
+
+class AzureFunctionsHook(HttpHook):
```

Comment:
There is official Python SDK https://pypi.org/project/azure-functions/
why do we not use it?

Reply:
I did check on this SDK, it supports to create azure-function, and deploy it, we cannot invoke the azure function which already exists.I am  building the HTTP request which can call Azure Function here

Reply:
@eladkal can you please check the above comment ^^

Reply:
is this by design or is it a missing feature in the SDK?
just wondering if we have a future todo to follow in the SDK issue list?

Reply:
I believe thats by design, as there are multiple ways to trigger Azure function. The simplest and most common use case is that of HttpTrigger.